### PR TITLE
♻️Maintenance: mypy simcore sdk

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
enable mypy in simcore-sdk

~~@pcrespov I need your help for this one:~~
```
Using mypy 1.2.0 (compiled: yes)
Mypy config /home/sanderegg/dev/github/osparc-simcore/mypy.ini
Mypying /home/sanderegg/dev/github/osparc-simcore/packages/simcore-sdk/src:
src/simcore_sdk/node_ports_v2/nodeports_v2.py:180:49: error: Argument "model" to "ValidationError" has incompatible type "Nodeports"; expected "Union[Type[BaseModel], Type[Dataclass]]"  [arg-type]
Found 1 error in 1 file (checked 24 source files)
```
fixed.

### CODE REVIEW

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py [hunk 1]
```diff
@@ -148,7 +147,7 @@ async def set_multiple(
 
         raises ValidationError
         """
-        tasks = deque()
+        tasks = []
```
Comment (GitHK) [inline packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py:150]:
```suggestion
        tasks: duque[Coroutine] = duque()
```

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port.py [hunk 2]
```diff
@@ -146,13 +146,17 @@ def __init__(self, **data: Any):
...
+            def _converter(value: ItemConcreteValue) -> ItemConcreteValue:
+                return value
+
+            self._py_value_converter = _converter
 
         else:
             assert self.property_type in TYPE_TO_PYTYPE  # nosec
 
-            self._py_value_type = TYPE_TO_PYTYPE[self.property_type]
+            self._py_value_type = (TYPE_TO_PYTYPE[self.property_type],)
```
Comment (GitHK) [inline packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port.py:159]: are you sure about this one?

**Review by GitHK — APPROVED** [review]
👍

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py (hunk 1 above)
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py:150]: Please read this (and there are several in google):
https://devpress.csdn.net/python/62f4ec94c6770329307faa09.htm

when appending, a list is on par with a deque. using a deque for no performance gain in a location that does not a priori need it, that gets then converted to a list is not something that makes the code clearer so I prefer not to do it.

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port.py (hunk 2 above)
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port.py:159]: well at least I can tell you that the way this was done before was wrong and not according to the type hints that are set on that variable

Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port.py:159]: actually wondering if that thing is even used... I will remove it

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

**Review by pcrespov — APPROVED** [review]
Amazing! :tada:  Greatly appreciated!

**Pull request merged**